honeybadgermpc/ntl/ctypes.pxd

```diff
@@ -1,41 +1,48 @@
 from libcpp.vector cimport vector
 
 cdef extern from "ntlwrapper.h":
-    cdef cppclass ZZ_c "ZZ":
+    cdef cppclass ZZ "ZZ":
+        ZZ() except +
         pass
 
     cdef cppclass ZZ_pContext_c "ZZ_pContext":
         void restore()
         pass
 
-    cdef cppclass ZZ_p_c "ZZ_p":
-        pass
+    cdef cppclass ZZ_p "ZZ_p":
+        ZZ_p() except +
 
     cdef cppclass ZZ_pX_c "ZZ_pX":
         void SetMaxLength(size_t)
         pass
 
     cdef cppclass vec_ZZ_p:
-        ZZ_p_c& operator[](size_t)
-        void SetLength(int n)
+        ZZ_p& operator[](size_t)
+        void SetLength(int n) nogil
         void kill()
         size_t length()
     cdef cppclass mat_ZZ_p:
         void SetDims(int m, int n)
         vec_ZZ_p& operator[](size_t)
         void kill()
 
-    void ZZ_p_init "ZZ_p::init"(ZZ_c x)
-    void SetNumThreads(int n)
-    void ZZ_conv_from_int "conv"(ZZ_c x, int i)
-    void ZZ_p_conv_from_int "conv"(ZZ_p_c x, int i)
-    void ZZ_conv_to_int "conv"(int i, ZZ_c x)
-    void ZZ_conv_from_long "conv"(ZZ_c x, long l)
-    void ZZ_conv_to_long "conv"(long l, ZZ_c x)
+
+    void ZZ_p_init "ZZ_p::init"(ZZ x) nogil
+    void SetNTLNumThreads_c "SetNumThreads"(int n)
+    void ZZonv_from_int "conv"(ZZ x, int i)
```

Comment:
ZZonv? Is there a reason to name these this way?

Reply:
Find-and-replace went wrong in a few places. These functions are not required anymore. Removing them.